2.0/IridescenceSuzanne/glTF/IridescenceSuzanne.gltf

```diff
@@ -0,0 +1,315 @@
+{
+    "asset" : {
+        "generator" : "Khronos glTF Blender I/O v1.1.45",
+        "version" : "2.0"
+    },
+    "extensionsRequired": [
+        "KHR_lights_punctual",
+        "KHR_materials_ior",
+        "KHR_materials_transmission",
+        "KHR_materials_volume",
+        "KHR_materials_iridescence"
+    ],
```

Comment:
None of these extensions ever belong in `extensionsRequired`, because none of them are needed to understand the basic geometry & file format of what's here.  Viewers can skip newer features and still load valid data from the core glTF format without these extensions.

"Required" extensions are things like Draco (when no fallback is present) and KTX2 (no fallback).

Reply:
@emackey I added two more examples to this PR which show iridescence effects on metallic/dielectic materials with varying parameters for (complex) material IOR, thin-film IOR and thin-film thickness.

Reply:
@PascalSchoen these models are great to have, I especially appreciate the 3-parameter grids of spheres. 

Could you explain the complex IOR parameter? How are those values being applied in the glTF materials? I can't seem to find them in IridescenceMetallicSpheres.gltf. 

Also it seems that KHR_materials_ior is not present in any of the materials in IridescenceMetallicSpheres.gltf, although it is defined in extensionsUsed.

Reply:
@echadwick-wayfair The complex IOR is calculated from the base color of the material using the artistic IOR function from this paper: http://jcgt.org/published/0003/04/03/paper.pdf

I added this to the implementation in the glTF Sample Viewer: https://github.com/PascalSchoen/glTF-Sample-Viewer/blob/feature/KHR_materials_iridescence/source/Renderer/shaders/functions.glsl#L87

It is completely debatable if this is the way to go. I just tried to be as close as possible to other implementations like in RenderMan without introducing another parameter to the KHR_materials_iridescence extension. Usually you would require a reflectivity (base color/F0) and edge tint parameter. I approximated edge tint with the base color for metals. By using this algorithm we can have three different complex IOR values - one for R, G and B.